### PR TITLE
GH-35943: [Dev] Ensure link issue works when PR body is empty

### LINKED ISSUE #35943: [Dev] Dev PR workflow fails to add link to GH issue if the body of PR is empty
### Describe the bug, including details regarding any error messages, version, and platform.

It has been brought up on Zulip that if the body of the PR is empty the Dev PR workflow fails and the link to the issue.

Examples:
* https://github.com/apache/arrow/pull/35835 should be linked to https://github.com/apache/arrow/issues/33980
* https://github.com/apache/arrow/pull/35933 should be linked to https://github.com/apache/arrow/issues/35599

### Component(s)

Developer Tools

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Fix a bug when PR body is empty and Dev workflow fails.

### What changes are included in this PR?

Ensure the link issue comment works in case of description being empty.

### Are these changes tested?

I have tested it on my fork here: https://github.com/raulcd/arrow/pull/81

### Are there any user-facing changes?

No
* Closes: #35943

### CODE REVIEW

File: .github/workflows/dev_pr/link.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ async function commentGitHubURL(github, context, pullRequestNumber, issueID) {
   const issueInfo = await helpers.getGitHubInfo(github, context, issueID, pullRequestNumber);
   const message = "* Closes: #" + issueInfo.number
   if (issueInfo) {
-    if (context.payload.pull_request.body.includes(message)) {
+    if (context.payload.pull_request.body != null && context.payload.pull_request.body.includes(message)) {
```
Comment (kou) [inline .github/workflows/dev_pr/link.js:87]: How about simplifying this?

```js
    if (context.payload.pull_request.body && context.payload.pull_request.body.includes(message)) {
```

or

```js
const body = context.payload.pull_request.body || "";
if (body.includes(message)) {
  return;
}
await github.rest.pulls.update({..., body: body + "\n" + message});
```
```

**Review by kou — APPROVED** [review]
+1

File: .github/workflows/dev_pr/link.js (hunk 1 above)
Comment (raulcd) [reply .github/workflows/dev_pr/link.js]: Thanks @kou ! I've applied the second suggested approach!

**Comment by raulcd** [comment]
:+1:

**Pull request merged**